otx/algorithms/segmentation/adapters/mmseg/configurer.py

```diff
@@ -0,0 +1,618 @@
+            self.configure_classes(cfg)
+            # Ignored mode
+            self.configure_ignore(cfg)
+
+    def configure_ignore(self, cfg):
+        """Change to incremental loss (ignore mode)."""
+        if cfg.get("ignore", False):
+            cfg_loss_decode = ConfigDict(
+                type="CrossEntropyLossWithIgnore",
+                use_sigmoid=False,
+                loss_weight=1.0,
+            )
+
+            if "decode_head" in cfg.model:
+                decode_head = cfg.model.decode_head
+                if decode_head.type == "FCNHead":
+                    decode_head.type = "CustomFCNHead"
+                    decode_head.loss_decode = cfg_loss_decode
+
+    # pylint: disable=too-many-branches
+    def configure_classes(self, cfg):
+        """Patch classes for model and dataset."""
+        org_model_classes = self.get_model_classes(cfg)
+        data_classes = self.get_data_classes(cfg)
+
+        if "background" not in org_model_classes:
+            org_model_classes = ["background"] + org_model_classes
+        if "background" not in data_classes:
+            data_classes = ["background"] + data_classes
+
+        # Model classes
+        if self.task_adapt_op == "REPLACE":
+            if len(data_classes) == 1:  # 'background'
+                model_classes = org_model_classes.copy()
+            else:
+                model_classes = data_classes.copy()
+        elif self.task_adapt_op == "MERGE":
+            model_classes = org_model_classes + [cls for cls in data_classes if cls not in org_model_classes]
+        else:
+            raise KeyError(f"{self.task_adapt_op} is not supported for task_adapt options!")
+
+        cfg.task_adapt.final = model_classes
+        cfg.model.task_adapt = ConfigDict(
+            src_classes=org_model_classes,
+            dst_classes=model_classes,
+        )
+
+        # Model architecture
+        if "decode_head" in cfg.model:
+            decode_head = cfg.model.decode_head
+            if isinstance(decode_head, dict):
+                decode_head.num_classes = len(model_classes)
+            elif isinstance(decode_head, list):
+                for head in decode_head:
+                    head.num_classes = len(model_classes)
+
+            # For SupConDetCon
+            if "SupConDetCon" in cfg.model.type:
+                cfg.model.num_classes = len(model_classes)
+
+        # Task classes
+        self.org_model_classes = org_model_classes
+        self.model_classes = model_classes
+
+    # Functions below are come from base stage
+    def configure_ckpt(self, cfg, model_ckpt):
+        """Patch checkpoint path for pretrained weight.
+
+        Replace cfg.load_from to model_ckpt
+        Replace cfg.load_from to pretrained
+        Replace cfg.resume_from to cfg.load_from
+        """
+        if model_ckpt:
+            cfg.load_from = self.get_model_ckpt(model_ckpt)
+        if cfg.get("resume", False):
+            cfg.resume_from = cfg.load_from
```

Comment:
We need code here to replace pretrained in the backbone with None if load_from and pretrained come in together, see classification.

Reply:
Thanks for nice catch, I'll update it.

Reply:
Updated.